tools/query_verification/src/main/java/com/google/bigquery/TeradataManager.java

```diff
@@ -0,0 +1,364 @@
+            } catch (SQLException e) {
+                // Print out errors returned from TD
+                jobResult = QueryJobResults.create(statement, query, e.getMessage(), null, null);
+            }
+
+            // Store results
+            jobResults.add(jobResult);
+        }
+
+        // Clear tables created
+        tables.forEach(this::deleteTable);
+
+        closeConnection();
+
+        return jobResults;
+    }
+
+    /**
+     * Reads connection properties to TD database from config file
+     * @return databaseServerName, username, password
+     */
+    private String[] getConnectionPropertiesFromConfigFile() {
+        InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream("config.json");
+        String configContents = new BufferedReader(new InputStreamReader(inputStream)).lines().collect(Collectors.joining("\n"));
+
+        JsonObject configJson = JsonParser.parseString(configContents).getAsJsonObject();
+        JsonObject tdJsonObject = configJson.getAsJsonObject(getName().toLowerCase());
+
+        String[] connectionProperties = new String[3];
+        connectionProperties[0] = tdJsonObject.get("databaseServerName").getAsString();
+        connectionProperties[1] = tdJsonObject.get("username").getAsString();
+        connectionProperties[2] = tdJsonObject.get("password").getAsString();
+
+        if (connectionProperties[0].isEmpty()) {
+            throw new IllegalArgumentException("Please enter the " + getName() + " database server name and credentials in config.json");
+        }
+
+        return connectionProperties;
+    }
+
+    /**
+     * Establishes a connection to the TD database
+     */
+    private void setupConnection() {
+        try {
+            String[] connectionProperties = getConnectionPropertiesFromConfigFile();
+            String url = "jdbc:teradata://" + connectionProperties[0] + "/TMODE=ANSI,CHARSET=UTF8";
+            String user = connectionProperties[1];
+            String password = connectionProperties[2];
+
+            // Create connection for running queries
+            Connection connection = DriverManager.getConnection(url, user, password);
+            statementConnection = connection.createStatement();
+
+            // Create connection for uploading CSV files
+            csvConnection = DriverManager.getConnection(url + ",TYPE=FASTLOADCSV", user, password);
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+    }
+
+    /**
+     * Closes the connection to the TD database
+     */
+    private void closeConnection() {
+        try {
+            statementConnection.close();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+    }
+
+    public List<String> getStatementsFromQuery() {
+        return Arrays.stream(query.query().split(";")).map(String::trim).filter(statement -> !statement.isEmpty()).collect(Collectors.toList());
```

Comment:
This isn't going to work if there's semicolons in quotes I think. Can you add a comment noting this?

Reply:
That's true, updated